tests/framework/script.go

```diff
@@ -0,0 +1,128 @@
+package framework
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+func RunScript(path string, args []string) ([]byte, error) {
+	args = append([]string{path}, args...)
+	return exec.Command("/bin/bash", args...).Output()
+}
+
+func RunScriptOutputToFile(scriptPath string, args []string, outputPath string, ci ClusterInfo) error {
+	output, err := RunScript(scriptPath, args)
+	if err != nil {
+		return fmt.Errorf("Received error running script %v: %w", scriptPath, err)
+	}
+
+	//nolint:gosec
+	resFile, err := os.OpenFile(outputPath, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0o777)
+	if err != nil {
+		return err
+	}
+	defer resFile.Close()
+
+	err = writeSystemInfoToFile(resFile, ci)
+	if err != nil {
+		return fmt.Errorf("Could not write system info to file %v: %w", outputPath, err)
+	}
+
+	if _, err := resFile.Write(output); err != nil {
+		return fmt.Errorf("Could not write script output to file %v: %w", outputPath, err)
+	}
+	return nil
+}
+
+func AddEntryToHostsFile(domain string, ip string) error {
+	hostsFile, err := os.OpenFile("/etc/hosts", os.O_APPEND|os.O_WRONLY, 0o644)
```

Comment:
I wonder our approach to these tests....because editing this file as part of automation feels too privileged.

Can we just utilize port forwarding? We'd be bypassing the LB, but we're still ultimately testing the traffic flow of our product.

Reply:
I am still utilising port forwarding when it's local, the issue I was getting though is that wrk wasn't connecting when using the IP directly instead of the hostname and it doesn't have any `--resolve` or similar option. Also, in the output, we don't see the hostname, just the IP:port. I can look again today. 

Having said that, I think the VM tests can still take this approach, as it's an ephemeral environment. These `wrk` tests don't make sense to run locally anyway, so maybe they can be skipped in those cases.

Reply:
Can we avoid using scripts and `wrk` and instead just use pure Go?

Reply:
> Can we just utilize port forwarding? We'd be bypassing the LB, but we're still ultimately testing the traffic flow of our product.

port-forwarding will put a proxy between wrk and data plane, which will affect performance numbers

> Can we avoid using scripts and wrk and instead just use pure Go? Having automation edit the /etc/hosts file, even if in an ephemeral environment, feels like a security issue waiting to happen.

what if we run it in a container? so we start wrk in a container with modified /etc/hosts

Reply:
> port-forwarding will put a proxy between wrk and data plane, which will affect performance numbers

If we go through the LB it's still the same issue though

Reply:
afaik, port forwarding works through Kubernetes API server through one of its go processes. So it is a constrained user land proxy.

Reply:
> Can we avoid using scripts and wrk and instead just use pure Go?

This might work: https://github.com/form3tech-oss/f1

I like the idea of using Go

Reply:
I looked a bit today and came across https://github.com/tsenart/vegeta - looked at f1 too, but I liked Vegeta because it provides a mechanism to provide a proxy URL which solves the /etc/hosts problem (plus it provides histogram output, ability to do distributed testing etc, which I'm not using now, but could be a nice addition later). Its output is a bit different than wrk so it's a bit of a departure - I've pushed up those results as well in the last commit.